hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/ReflectiveCompactSerializer.java

```diff
@@ -152,7 +169,8 @@ public Object read(@Nonnull CompactReader reader) throws IOException {
     @Nonnull
     private Object createObject(Class associatedClass) {
         try {
-            return ClassLoaderUtil.newInstance(associatedClass.getClassLoader(), associatedClass);
+            ObjectInstantiator instantiator = objectInstantiators.computeIfAbsent(associatedClass, objenesis::getInstantiatorOf);
```

Comment:
We don't allow dependencies at hazelcast. I would use objenesis as well if that is not the case.